platform/linux-generic/odp_packet.c

```diff
@@ -1953,6 +1953,7 @@ static inline uint8_t parse_ipv4(packet_parser_t *prs, const uint8_t **parseptr,
 		if (chksum_finalize(chksum_partial(ipv4, ihl * 4, 0)) != 0xffff) {
 			prs->flags.ip_err = 1;
 			prs->flags.l3_chksum_err = 1;
+			*offset += ihl * 4;
```

Comment:
Patch v5 which included bad ckcum check for IPv4 header broke pktio validation test in linux_gen platform. Upon inspecting logs, there was failure at test/validation/api/pktio/pktio.c:229 as head.magic != TEST_SEQ_MAGIC. It was due to incorrect L4 offset used to retrieve packet data for this check, which arises from odp_packet.c:2181 where parse_ipv4() did not set offset pointer (for L4) when L3 checksum was incorrect in the received packet and the former pointer value will eventually be stored in packet_parser->l4_offset (and consequently, this will be used to retrieve packet payload following L4).

I agree when checksum fails, ihl could be corrupt, we anyway will have odp_packet_l3_chksum_status() returning ODP_PACKET_CHKSUM_BAD which is actually the test here. So, this patch is to enable the test do its validation part successfully.

Reply:
Good explanation, thank you. So of course in case of incorrect L3 checksum, the L4 checks shouldn't be done. I'll comment the other commit. Also, I think in that case the l4_offset should be invalid, but currently it isn't. So instead of this change, the caller (packet_parse_common_l3_l4()) should set l4_offset only if parse_ipv4() returns success.

Reply:
When L3 checksum failed, implementation cannot be sure about correct L4 offset. So, it's better to set it to ODP_PACKET_OFFSET_INVALID. API does not guarantee to application that, but it's safer to set it invalid in a software parser while it does not harm performance, etc. With HW parsers the offset may not be set to invalid without performance cost, so API leaves it open.

Reply:
Set to invalid offset, instead of leaving it to L3 offset or incrementing to L4 offset.
`*offset =ODP_PACKET_OFFSET_INVALID;`

Reply:
@psavol Apologies for a quite longer delay in response. I have followed your and @JereLeppanen suggestions and sent v9 patch. Please check if the commits are good. Appreciating both of your time.